event-handler/teleport_event.go

```diff
@@ -67,109 +63,62 @@ type TeleportEvent struct {
 	}
 }
 
-// printEvent represents an artificial print event struct which adds json-serialisable data field
-type printEvent struct {
-	EI          int64     `json:"ei"`
-	Event       string    `json:"event"`
-	Data        []byte    `json:"data"`
-	Time        time.Time `json:"time"`
-	ClusterName string    `json:"cluster_name"`
-	CI          int64     `json:"ci"`
-	Bytes       int64     `json:"bytes"`
-	MS          int64     `json:"ms"`
-	Offset      int64     `json:"offset"`
-	UID         string    `json:"uid"`
-}
-
 // NewTeleportEvent creates TeleportEvent using AuditEvent as a source
-func NewTeleportEvent(e events.AuditEvent, cursor string) (*TeleportEvent, error) {
+func NewTeleportEvent(e *auditlogpb.EventUnstructured, cursor string) (*TeleportEvent, error) {
+	payload, err := e.Unstructured.MarshalJSON()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 	evt := &TeleportEvent{
 		Cursor: cursor,
 		Type:   e.GetType(),
-		Time:   e.GetTime(),
+		Time:   e.Time.AsTime(),
 		Index:  e.GetIndex(),
+		ID:     e.Id,
```

Comment:
There was some extra logic when setting the `id`. Is it no longer required?

Reply:
That logic was moved to auth server